src/contract/assembled_transaction.ts

```diff
@@ -836,28 +846,40 @@ export class AssembledTransaction<T> {
 
     // eslint-disable-next-line no-restricted-syntax
     for (const [i, entry] of authEntries.entries()) {
+      // workaround for https://github.com/stellar/js-stellar-sdk/issues/1070
+      const credentials = xdr.SorobanCredentials.fromXDR(entry.credentials().toXDR())
       if (
-        entry.credentials().switch() !==
+        credentials.switch() !==
         xdr.SorobanCredentialsType.sorobanCredentialsAddress()
```

Comment:
if you add `.value` here it will also work around the above issue since you'd be comparing integers

Reply:
If you do that it will fail on the 
```ts
const authEntryAddress = Address.fromScAddress(
  credentials.address().address(),
).toString();
```
line. For reasons I do not understand but they are the same reasons for why 
```
const credentials = xdr.SorobanCredentials.fromXDR(entry.credentials().toXDR())
```
is needed in the first place. Because™

Reply:
Brilliant :laughing: then this'll have to do